server/src/jira/token.guard.ts

```diff
@@ -0,0 +1,14 @@
+import { Injectable, ExecutionContext, CanActivate, UnauthorizedException } from "@nestjs/common";
+import { Request } from "express";
+import { SessionData } from "express-session";
+
+@Injectable()
+export class SessionTokenGuard implements CanActivate {
+  async canActivate(context: ExecutionContext) {
+    const req: Request = context.switchToHttp().getRequest();
+    const session: SessionData = req.session;
```

Comment:
If you use a type parameter to type this, it becomes slightly cleaner and you don't have to coerce types which is a big plus:

```
    const request = context.switchToHttp().getRequest<Request>();
    const session = request.session;
```